src/drivers/intel/ssp/ssp.c

```diff
@@ -67,22 +67,23 @@ static void ssp_empty_tx_fifo(struct dai *dai)
 /* empty SSP receive FIFO */
 static void ssp_empty_rx_fifo(struct dai *dai)
 {
+	struct timer *timer = timer_get();
+	uint64_t deadline = platform_timer_get(timer) +
+		clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1) *
+		SSP_RX_FLUSH_TIMEOUT / 1000;
 	uint32_t sssr;
-	uint32_t entries;
-	uint32_t i;
 
-	sssr = ssp_read(dai, SSSR);
+	do {
+		while (ssp_read(dai, SSSR) & SSSR_RNE)
+			/* read to empty the fifo */
+			ssp_read(dai, SSDR);
```

Comment:
We should care here about infinite loop, shouldn't we ? Eg. in case of some SSP HW failure or wrong initialization.

Reply:
Use the same deadline in this condition here as well?

Reply:
It should be fine

Reply:
We should trust that the SSP basic logic work, ssp_read() will read out entries rapidly so the SSSR_RNE will become 0 soon, adding deadline here as well is overkill IMO.